src/index.js

```diff
@@ -49,9 +50,13 @@ class CID {
    * new CID(<cid.buffer>)
    * new CID(<multihash>)
    * new CID(<bs58 encoded multihash>)
+   * new CID(<CID>)
    *
    */
   constructor (version, codec, multihash) {
+    if (CID.isCID(version)) {
+      return version
```

Comment:
Yup. You're right. Side effects are not good. Maybe something like this could fit.
But I'm not sure it works. I need to study more about multiformats.

```js
constructor (version, codec, multihash) {
    if (CID.isCID(version)) {
      let cid = version
      let version = cid.version
      let codec = cid.codec
      let multihash = Buffer.from(cid.multihash)
      return new CID(version, codec, multihash)
    }
    ...
```